lib/MySQL_Thread.cpp

```diff
@@ -3558,90 +3271,35 @@ MySQL_Session * MySQL_Thread::create_new_session_and_client_data_stream(int _fd)
-	}
-	sess->client_myds->myconn->options.session_track_gtids=strdup(mysql_thread___default_session_track_gtids);
-
-	uint32_t sql_auto_is_null_int=SpookyHash::Hash32(mysql_thread___default_sql_auto_is_null,strlen(mysql_thread___default_sql_auto_is_null),10);
-	sess->client_myds->myconn->options.sql_auto_is_null_int = sql_auto_is_null_int;
-	if (sess->client_myds->myconn->options.sql_auto_is_null) {
-		free(sess->client_myds->myconn->options.sql_auto_is_null);
-	}
-	sess->client_myds->myconn->options.sql_auto_is_null=strdup(mysql_thread___default_sql_auto_is_null);
-
-	uint32_t sql_select_limit_int=SpookyHash::Hash32(mysql_thread___default_sql_select_limit,strlen(mysql_thread___default_sql_select_limit),10);
-	sess->client_myds->myconn->options.sql_select_limit_int = sql_select_limit_int;
-	if (sess->client_myds->myconn->options.sql_select_limit) {
-		free(sess->client_myds->myconn->options.sql_select_limit);
-	}
-	sess->client_myds->myconn->options.sql_select_limit=strdup(mysql_thread___default_sql_select_limit);
-
-	uint32_t sql_safe_updates_int=SpookyHash::Hash32(mysql_thread___default_sql_safe_updates,strlen(mysql_thread___default_sql_safe_updates),10);
-	sess->client_myds->myconn->options.sql_safe_updates_int = sql_safe_updates_int;
-	if (sess->client_myds->myconn->options.sql_safe_updates) {
-		free(sess->client_myds->myconn->options.sql_safe_updates);
-	}
-	sess->client_myds->myconn->options.sql_safe_updates=strdup(mysql_thread___default_sql_safe_updates);
-
-	uint32_t collation_connection_int=SpookyHash::Hash32(mysql_thread___default_collation_connection,strlen(mysql_thread___default_collation_connection),10);
-	sess->client_myds->myconn->options.collation_connection_int = collation_connection_int;
-	if (sess->client_myds->myconn->options.collation_connection) {
-		free(sess->client_myds->myconn->options.collation_connection);
-	}
-	sess->client_myds->myconn->options.collation_connection=strdup(mysql_thread___default_collation_connection);
-
-	uint32_t net_write_timeout_int=SpookyHash::Hash32(mysql_thread___default_net_write_timeout,strlen(mysql_thread___default_net_write_timeout),10);
-	sess->client_myds->myconn->options.net_write_timeout_int = net_write_timeout_int;
-	if (sess->client_myds->myconn->options.net_write_timeout) {
-		free(sess->client_myds->myconn->options.net_write_timeout);
-	}
-	sess->client_myds->myconn->options.net_write_timeout=strdup(mysql_thread___default_net_write_timeout);
-
-	uint32_t max_join_size_int=SpookyHash::Hash32(mysql_thread___default_max_join_size,strlen(mysql_thread___default_max_join_size),10);
-	sess->client_myds->myconn->options.max_join_size_int = max_join_size_int;
-	if (sess->client_myds->myconn->options.max_join_size) {
-		free(sess->client_myds->myconn->options.max_join_size);
+	for (int i=0; i<SQL_NAME_LAST; i++) {
+		if (i == SQL_CHARACTER_SET || i == SQL_CHARACTER_SET_RESULTS ||
+			i == SQL_CHARACTER_SET_CONNECTION || i == SQL_CHARACTER_SET_CLIENT ||
+			i == SQL_CHARACTER_SET_DATABASE) {
```

Comment:
All these variables cannot be set independently.

https://dev.mysql.com/doc/refman/8.0/en/charset-connection.html#charset-connection-client-configuration

We either:
* only set `collation_connection` (what is that is actually exchanged in the protocol)
* set `collation_connection` , and all the other variables need to be set accordingly

Reply:
Default character set is used when backend mysql server dose not support configured collation/charset. For example, mysql 5.7 does not support collation is greater than 254. There for it might be configured to switch to the default character set.

This code is useless. I should not add it, because default values should not be used to initialize variables.